nearcore/src/config_validate.rs

```diff
@@ -33,6 +33,14 @@ impl<'a> ConfigValidator<'a> {
     fn validate_all_conditions(&mut self) {
         self.validate_cold_store_config();
         self.validate_cloud_archival_config();
+        self.validate_tracked_shards_config();
+
+        if let Some(state_sync) = &self.config.state_sync {
+            if let Some(dump_config) = &state_sync.dump {
+                self.validate_state_dumper_config(dump_config);
+            }
```

Comment:
optional nit: since dump is part of state_sync config, consider moving this into validate_state_sync_config function.